### PR TITLE
Feat/create post

### PR DESCRIPTION
**Pull request by jeromes123**
**OVERVIEW**
Created create post functionality with media upload.

**SNIPPETS**
image

### CODE REVIEW

File: src/resources/views/components/create-post.blade.php [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+                <a href="/"><img src="assets/images/logo.png" alt="homepage"></a>
+            </div>
+            <ul class="navigation">
+                <a href=""><li>My Profile</li></a>
+                <a href=""><li><span>⌂</span></li></a>
+                <a href=""><li><span>☰</span></li></a>
+            </ul>
+        </div>
+    </header>
+    -->
```
Comment (mikco-hipe) [inline src/resources/views/components/create-post.blade.php]: remove if not needed

File: src/routes/web.php [hunk 2]
```diff
@@ -58,3 +61,17 @@
...
 ]);
+
+Route::get('/create-post', function() {
+    $userPosts = UserPost::all();
+    $postMedia = PostMedia::all();
+    return view('components.create-post', [
+        'userPosts' => $userPosts, 
+        'postMedia' => $postMedia
+    ]);
+});
```
Comment (mikco-hipe) [inline src/routes/web.php]: Im not sure if this is correct but If you want to display posts, you need to create another route and handle that in the controller. 

This route is for "GET" method but your route is for creating post, creating posts is for POST method

File: src/resources/views/components/create-post.blade.php [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    <tbody>
+        @foreach ($userPosts as $post)
+        <tr>
+            <td>{{ $post->user_id }}</td>
+            <td>{{ $post->content }}</td>
+            <td>
+                @if (in_array($post->id, $postMedia->pluck('post_id')->toArray()))
+                    @php
+                        $postMedia = App\Models\PostMedia::where('post_id', $post->id)->first();
+                    @endphp
```
Comment (mikco-hipe) [inline src/resources/views/components/create-post.blade.php]: now that you have dedicated route for displaying the posts, you dont need this. instead call `postMedia `

File: src/app/Http/Controllers/UserPostController.php [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    }
+
+    /**
+     * Gets all the posts and media
+     * Displays the posts page
+     */
+    public function showPostsPage(): View
+    {
+        $userPosts = UserPost::all();
+        $postMedia = PostMedia::all();
```
Comment (mikco-hipe) [inline src/app/Http/Controllers/UserPostController.php]: transfer to service

File: src/app/Services/UserPostService.php [hunk 5]
```diff
@@ -33,4 +34,33 @@ public function create(CreateUserPostRequest $request): void
...
+            } else {
+                // If no media exists for this post, add only the post to the array
+                $postsAndMedia[] = [
+                    'post' => $post,
+                    'postMedium' => null
+                ];
+            }
+        }
+        return $postsAndMedia; // Convert array to Collection and return
+    }
```
Comment (mikco-hipe) [inline src/app/Services/UserPostService.php]: This can be optimized

File: src/routes/web.php [hunk 6]
```diff
@@ -58,3 +61,13 @@
...
+
+Route::get('/posts-page', [
+    UserPostController::class,
+    'showPostsPage'
+]);
+
+Route::post('/create-post', [
+    UserPostController::class, 
+    'create'
+]);
```
Comment (mikco-hipe) [inline src/routes/web.php]: I noticed there's no middleware on these. Does this mean even unauthenticated user can post and view posts?

**Review by mikco-hipe — COMMENTED** [review]
Please check!

File: src/routes/web.php (hunk 6 above)
Comment (jeromes123) [reply src/routes/web.php]: Unauthenticated users must not view or create posts. I will add middleware.

**Review by mikco-hipe — APPROVED** [review]
LGTM 🐐

**Review by mikco-hipe — APPROVED** [review]

**Pull request merged**